### PR TITLE
Created ObsoleteByRefModifier inspection

### PR DESCRIPTION
**Pull request by rkapka**
This is my first contribution, so please be gentle ;-)

### CODE REVIEW

File: RetailCoder.VBE/Inspections/ObsoleteByRefModifierInspection.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+using Rubberduck.Common;
+using Rubberduck.Inspections.Abstract;
+using Rubberduck.Inspections.Resources;
+using Rubberduck.Inspections.Results;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+
+namespace Rubberduck.Inspections
+{
+    public sealed class ObsoleteByRefModifierInspection : InspectionBase
```
Comment (retailcoder) [inline RetailCoder.VBE/Inspections/ObsoleteByRefModifierInspection.cs]: "Obsolete" is a bit of a strong word for a keyword that's completely legit and has no alternative - how about renaming that to `RedundantByRefModifierInspection`?

Comment (retailcoder) [inline RetailCoder.VBE/Inspections/ObsoleteByRefModifierInspection.cs]: I wonder if that couldn't be implemented as a parse tree inspection...

File: RetailCoder.VBE/Inspections/QuickFixes/RemoveExplicitByRefModifierQuickFix.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+using Rubberduck.VBEditor;
+using Rubberduck.Parsing;
+using Rubberduck.Parsing.Grammar;
+using System.Collections.Generic;
+using Rubberduck.Parsing.Symbols;
+
+namespace Rubberduck.Inspections.QuickFixes
+{
+    public class RemoveExplicitByRefModifierQuickFix : QuickFixBase
+    {
```
Comment (retailcoder) [inline RetailCoder.VBE/Inspections/QuickFixes/RemoveExplicitByRefModifierQuickFix.cs]: I think there needs to be a `SupportedInspections` override here

File: RetailCoder.VBE/Inspections/Results/ObsoleteByRefModifierInspectionResult.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+using Rubberduck.Inspections.Abstract;
+using Rubberduck.Inspections.QuickFixes;
+using Rubberduck.Inspections.Resources;
+using Rubberduck.Parsing.Symbols;
+
+namespace Rubberduck.Inspections.Results
+{
+    public class ObsoleteByRefModifierInspectionResult : InspectionResultBase
+    {
+        private readonly Lazy<IEnumerable<QuickFixBase>> _quickFixes;
```
Comment (retailcoder) [inline RetailCoder.VBE/Inspections/Results/ObsoleteByRefModifierInspectionResult.cs]: Aaah, that's how we did it prior to @Hosch250's "big dig" inspections/quick-fix refactoring. There's no more inspection-specific "result" class, and results are completely decoupled from quick-fixes now. hmm...

File: RetailCoder.VBE/Inspections/ObsoleteByRefModifierInspection.cs (hunk 1 above)
Comment (Hosch250) [inline RetailCoder.VBE/Inspections/ObsoleteByRefModifierInspection.cs]: Definitely.

**Pull request closed without merging**